jwst/lib/set_telescope_pointing.py

```diff
@@ -1,27 +1,73 @@
 """Set Telescope Pointing from quaternions"""
-from copy import copy
-import logging
-from math import (cos, sin)
-import os.path
-import sqlite3
+import sys
+
+import asdf
 from collections import defaultdict, namedtuple
+from copy import copy, deepcopy
+import dataclasses
+from enum import Enum
+import logging
+from math import (asin, atan2, cos, sin, sqrt)
+import typing
 
 from astropy.time import Time
-from scipy.interpolate import interp1d
 import numpy as np
+from scipy.interpolate import interp1d
 
+from .exposure_types import IMAGING_TYPES, FGS_GUIDE_EXP_TYPES
+from .set_velocity_aberration import compute_va_effects_vector
+from .siafdb import SIAF, SiafDb
 from ..assign_wcs.util import update_s_region_keyword, calc_rotation_matrix
 from ..assign_wcs.pointing import v23tosky
-from ..datamodels import Level1bModel
+from ..datamodels import open as dm_open
```

Comment:
Ah, if the raw level 1 files do not get DATAMODL, then yes, this should get backed out.

Reply:
As far as i know the only code that does anything with DATAMODL is ours (cal pipeline) and since this is the first time any of our datamodels touch a level-1 product there won't be a DATAMODL in the input (but there will be in the output).